standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -5224,6 +5233,7 @@ public void performTimeOuts() {
           }
         }
         LOG.info("Aborted " + numTxnsAborted + " transactions due to timeout");
+        Metrics.getOrCreateCounter(MetricsConstants.TOTAL_NUM_TIMED_OUT_TXNS).inc(numTxnsAborted);
```

Comment:
qq, should we report only if metrics are enabled?

Reply:
Metrics are only initialized if metrics are enabled. So here if they aren't enabled a dummy counter would be used I think.